src/device.rs

```diff
@@ -156,9 +156,8 @@ impl Device {
                     Err(e) => {
                         if controls.is_empty() || e.kind() != io::ErrorKind::InvalidInput {
                             return Err(e);
-                        } else {
-                            break;
                         }
+                        break;
                     }
```

Comment:
We could've almost written something like `break if ... { Err(e) } else { Ok(controls) };` showing that this is the only exit.

Reply:
Woah, that works just fine! I'm somewhat unfamiliar with this syntax, so please do check to ensure it's right:
```suggestion
                    Err(e) => {
                        break if controls.is_empty() || e.kind() != io::ErrorKind::InvalidInput {
                            Err(e)
                        } else {
                            Ok(controls)
                        };
```